### ISSUE TITLE
Identical values not equal when using path selector

### ISSUE THREAD

**Issue by untitledlt** [CacheControl/json-rules-engine#169]
Not sure what i'm doing wrong here but i can not make two identical values truthy.
Here is my rule:
```
conditions: {
        all: [{
            fact: 'root',
            path: '$.some.nested.object.changeTime',
            operator: 'equal',
            value: 1574172807460,
        }]
    },
```

Here is a debug log:
```
engine::run started
engine::run runtimeFacts:
almanac::constructor initialized runtime fact:root with [object Object]<object>
condition::evaluate extracting object property $.some.nested.object.changeTime
condition::evaluate <1574172807460 equal 1574172807460?> (false)
rule::evaluateConditions results
engine::run ruleResult:false
engine::run completed
almanac::factValue cache miss for fact:success-events; calculating
```

I tried to value to `'1574172807460'` as string and still getting `(false)`.
The same is with boolean value `<true equal true?> (false)`

If i use `notEqual` operator, it evaluates truthy.

What am i doing wrong here?

**Comment by MattCopenhaver** [comment]
I tried this, and it seems to work as expected for me.  Any other information you could provide?

**Comment by untitledlt** [comment]
Here is a minimal setup:
https://pastebin.com/mHix8P7N

```                                    
engine::addOperator name:equal
engine::addOperator name:notEqual
engine::addOperator name:in
engine::addOperator name:notIn
engine::addOperator name:contains
engine::addOperator name:doesNotContain
engine::addOperator name:lessThan
engine::addOperator name:lessThanInclusive
engine::addOperator name:greaterThan
engine::addOperator name:greaterThanInclusive
engine::run started
engine::run runtimeFacts:
almanac::constructor initialized runtime fact:first with [object Object]<object>
almanac::constructor initialized runtime fact:success-events with undefined<undefined>
condition::evaluate extracting object property $.second.third
condition::evaluate <true equal true?> (false)
rule::evaluateConditions results
engine::run ruleResult:false
engine::run completed
almanac::factValue cache miss for fact:success-events; calculating
```

**Comment by untitledlt** [comment]
I tried to downgrade to @4.1.0.
Now I'm getting:
```
almanac::constructor initialized runtime fact:first with [object Object]<object>
almanac::constructor initialized runtime fact:success-events with undefined<undefined>
condition::evaluate extracting object property $.second.third, received: undefined
condition::evaluate <undefined equal true?> (false)

```

**Comment by untitledlt** [comment]
I traced down this issue. 
Looks like it's comparing `[leftHandSideValue] === rightHandSideValue`

![image](https://user-images.githubusercontent.com/7152884/69165141-c2d38d00-0af9-11ea-826f-c04f1d82cd0f.png)

Any ideas why `leftHandSideValue` is taken as array?

**Comment by CacheControl** [comment]
@untitledlt I think the issue is your `path` starts with `$`, which is not a valid path. See the [docs](https://github.com/CacheControl/json-rules-engine/blob/master/docs/rules.md#condition-helpers-path) on path syntax. You likely need to change that to be `.some.nested.object.changeTime`

**Comment by untitledlt** [comment]
I tried that but it complains about `selectn`/`JsonPath` syntax:

```
engine::addOperator name:equal
engine::addOperator name:notEqual
engine::addOperator name:in
engine::addOperator name:notIn
engine::addOperator name:contains
engine::addOperator name:doesNotContain
engine::addOperator name:lessThan
engine::addOperator name:lessThanInclusive
engine::addOperator name:greaterThan
engine::addOperator name:greaterThanInclusive
engine::run started
engine::run runtimeFacts:
almanac::constructor initialized runtime fact:first with [object Object]<object>
almanac::constructor initialized runtime fact:success-events with undefined<undefined>
Oops! Looks like you're trying to use the deprecated syntax for the ".path" property.
Please convert your "path" properties to JsonPath syntax (ensure your path starts with "$")
Alternatively, if you wish to continue using old syntax (provided by selectn), you may "npm install selectn" as a direct dependency.
See https://github.com/CacheControl/json-rules-engine/blob/master/CHANGELOG.md#500--2019-10-27 for more information.
engine::run completed
almanac::factValue cache miss for fact:success-events; calculating
(node:47921) UnhandledPromiseRejectionWarning: Error: json-rules-engine: Unmet peer dependency "selectn" required for use of deprecated ".path" syntax. please "npm install selectn" or convert to json-path syntax
    at Almanac.factValue (/Users/untitled/workplace/smarthome.lt/daemon/node_modules/json-rules-engine/dist/almanac.js:170:19)
    at /Users/untitled/workplace/smarthome.lt/daemon/node_modules/json-rules-engine/dist/condition.js:135:24
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:745:11)
    at startup (internal/bootstrap/node.js:282:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
(node:47921) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:47921) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```

**Comment by untitledlt** [comment]
I'm using `5.0.0-alpha3` as it gets installed by default.

**Comment by chemdrew** [comment]
I'm getting this issue as well. It looks like with v5 the json path is returning the facts and values as arrays, which aren't the same reference so they fail the === check

here's a temporary workaround I used to test with
```javascript
engine.addOperator('equal', (factValue, jsonValue) => {
      console.log(factValue); // [ 'my-value' ]
      console.log(jsonValue); // [ 'my-value' ]
      return factValue[0] === jsonValue[0];
});
```

might also be worth noting that I am using async fact enrichers to get the value for it

**Comment by CacheControl** [comment]
@untitledlt @chemdrew This was a bug in the v5-alpha candidate. It has been fixed in **v5.0.0**
